pkg/utils/cloudconfig/azure/config.go

```diff
@@ -88,7 +88,7 @@ func (cfg *CloudConfig) validate() error {
 		return fmt.Errorf("resource group is empty")
 	}
 
-	if cfg.VnetName == "" {
+	if cfg.UserAgent == "fleet-member-agent" && cfg.VnetName == "" {
```

Comment:
Please move the vnetName check to where we new the config in the fleet repo.

As a common package, we should only validate the required the field for all the agents.
Any special check should not be hard-coded here.